lib/api/utilities/text_renderer.rb

```diff
@@ -39,7 +39,11 @@ def initialize(text, format: nil, object: nil)
         @text = text
         @format = format
         @object = object
-        @project = object.project if object.respond_to?(:project)
+        if object.respond_to?(:project)
+          @project = object.project
+        else
+          @project = object
```

Comment:
That was quite confusing at first, but refreshing in the end. Because IIRC [this](https://github.com/opf/openproject/pull/2990/files#diff-7c7091efdf7a63582087785ae217bf49L54) has been the only necessary change. But I think this is also covered in a separate commit, should be easy to spot.